### DISCUSSION TITLE
Erro na publicação do pacote privado

### DISCUSSION THREAD

**Discussion by LeonardoGuarilha** [balta-io/forum#84] [Tira Dúvidas] [answered]
CURSO: Fundamentos do Azure, Git, GitHub e DevOps

Ao tentar fazer a publicação do segundo pacote privado que depende do primeiro criado no curso, está dando o seguinte erro:
![image](https://user-images.githubusercontent.com/39388688/200383193-72c7c574-a579-487c-8596-30080e5ab3a3.png)
O meu main.yml está da seguinte forma
![image](https://user-images.githubusercontent.com/39388688/200383301-8f284686-9594-4db3-84af-23708a94fe72.png)
Verifiquei no repo do curso que está diferente a senha, está da seguinte forma
![image](https://user-images.githubusercontent.com/39388688/200383436-65282727-5aa6-4b31-aea9-20c8718c980d.png)

Esse "GH_PACK_SECRET" é alguma secret no repositório do git? Pois se eu só colocar ele no meu arquivo main.yml não funciona também

**Comment by andrebaltieri** [comment, marked as answer]
@LeonardoGuarilha me parece que ele não encontrou o pacote com o nome `MyCustomPrivatePackage`... Se fosse algo com token, acredito que seria outro erro, como 403...

Consegue verificar se o nome é este mesmo?

**Reply by LeonardoGuarilha** [reply]
O pacote está com o nome correto, conferi aqui.
Achei uma Discussion no Github sobre isso: https://github.com/orgs/community/discussions/30746
O problema era o mesmo, a solução foi criar um PAT e colocar no lugar do secrets.GITHUB_TOKEN. Não sei se isso teria algum impacto a nível de segurança.

**Reply by andrebaltieri** [reply]
Show!!! Obrigado por compartilhar 💜

**Reply by RafaelDifini** [reply]
> O pacote está com o nome correto, conferi aqui. Achei uma Discussion no Github sobre isso: https://github.com/orgs/community/discussions/30746 O problema era o mesmo, a solução foi criar um PAT e colocar no lugar do secrets.GITHUB_TOKEN. Não sei se isso teria algum impacto a nível de segurança.

pode mandar um print @LeonardoGuarilha , lendo os artigos e a documentação eu nao entendi muito bem, um PAT nao é oque ja criamos pro %appdata%\NuGet\NuGet.Config?

**Reply by LeonardoGuarilha** [reply]
@RafaelDifini, isso, é o PAT que foi criado para o NuGet. No meu caso eu só coloquei direto no main.yml mas creio que pode ser criado uma secret no repositório para isso.

![image](https://user-images.githubusercontent.com/39388688/201211661-48c3b10c-181f-4fc2-b37f-4fc9b1092fcc.png)

**Reply by RafaelDifini** [reply]
o mesmo que esta no nosso nuget entao ne amigo? entrre {}? coloquei ele só entre chaves e deu esse erro:
![image](https://user-images.githubusercontent.com/79666617/201213385-79c46470-9678-4afd-9cc4-58211c3dd21d.png)

**Reply by RafaelDifini** [reply]
Consegui amigo, a sintaxe é sem ${{    }}, obrigado!

**Reply by LeonardoGuarilha** [reply]
Isso, o mesmo do nuget eu coloquei no arquivo. O meu eu não coloquei em secret, então eu coloquei o token mesmo no -p sem "{}"
